protocol/urls/urls.go

```diff
@@ -140,6 +212,12 @@ func GetLinkPreviewData(link string) (previewData LinkPreviewData, err error) {
 	if "github.com" == hostname {
```

Comment:
Probably at this point a `switch/case` statement is more idiomatic.

I think we can have a single switch statement for all:
```
switch hostname {
   case "github.com": 
    ....
    case "giphy": 
    ...
    case "tenor.com":
    ...
    case "youtube.com", "youtube.be", ....:
     ... 
    default:
     	return previewData, fmt.Errorf("Link %s isn't whitelisted. Hostname - %s", link, url.Hostname())
}

```
Since all the cases in the list are actually well defined, we don't need anymore to iterate on the list

Reply:
I have made the change, the tests pass, but can you still check it once?
If it's okay, please resolve this comment.